### ISSUE TITLE
Cannot scaffold node-webkit due to rx.js with yo 1.8.4

### ISSUE THREAD

**Issue by neloaguila** [Dica-Developer/generator-node-webkit#86]
After executing `yo node-webkit` and answering the preceding questions until the version of node-webkit to download, the process suddenly stops, ending the execution with

```
   .../node_modules/yo/node_modules/rx/dist/rx.js:77
               throws e;
               ^
   true
```

I have searched the net for possible solutions but haven't found any. Can someone please help with this problem? Thank you very much.

**Comment by mschaaf** [comment]
Could you please provide the exact steps to reproduce. Thanks.

**Comment by neloaguila** [comment]
Okay, so first, I have installed generator-node-webkit globally with `npm install -g generator-node-webkit` giving me this
![screenshot from 2016-07-11 11-45-02](https://cloud.githubusercontent.com/assets/6396389/16719620/424f2a6a-475f-11e6-88c4-93e03003637b.png)

After that, I went to my app directory to scaffold node-webkit using yo
![screenshot from 2016-07-11 11-47-01](https://cloud.githubusercontent.com/assets/6396389/16719566/9194be24-475e-11e6-83d8-b6fe24636525.png)

And that's it.

I also tried `yo doctor` to see if somethings wrong with yo but no error was found.

Anyway, before all of these, I have received permission errors whenever I try to install npm packages globally so I created directory `.npm-global` within my home directory, changed npm prefix to `$HOME/.npm-global` and modified my $PATH to include `$HOME/.npm-global/bin`.

P.S. `$HOME = /home/nelo`

I hope you could find a way around this. Thanks!

**Comment by mschaaf** [comment]
First I can't reproduce your problem with the steps from above.

From the second screen there is a step missing that asks for the OS version to download.

You talked about permission problems. I recommend to clean all your users npm folders and start over. You also should consider to use nvm (https://github.com/creationix/nvm) to manage your node and npm installations.

What are the yo, npm and node versions you are using?

On my system it is:
node: v5.5.0
npm: 3.3.12
yo: 1.6.0

**Comment by neloaguila** [comment]
Yes, when scaffolding node-webkit, it never gets pass the step where the version to download is asked. After that question, it just stops with that rx.js error.

I did what you recommended. I've removed node, nvm and node_modules in my machine and started over. I've used nvm as you suggested and installed the latest version of node. Then I installed yo and generator-node-webkit, tried scaffolding node-webkit again and the error is still there.

Now I have the following versions:
node: v6.3.0
npm: 3.10.3
yo: 1.8.4

I guess the problem is with the latest version of yo, not of node-webkit.

**Comment by mschaaf** [comment]
I see I can reproduce with yo version 1.8.4. Is downgrading to version 1.6.x for now an option?

**Comment by neloaguila** [comment]
I've downgraded yo to version 1.6.0 and tried scaffolding with these versions:
node: v6.3.0
npm: 3.10.3
yo: 1.6.0
But this still results with the same error.

Then I also tried shifting to the versions that you have but the error still arises.
![screenshot from 2016-07-11 19-22-47](https://cloud.githubusercontent.com/assets/6396389/16729171/4d126e06-479d-11e6-9a61-e2b58ed7a741.png)

**Comment by neloaguila** [comment]
What if the problem is related to the node-webkit prompts? I have found that the rx module is required by another module needed by yo, the inquirer module. I have also found [this](https://github.com/SBoudrias/Inquirer.js/issues/363), which seems to have the same problem as this.

What do you think?

**Comment by mschaaf** [comment]
I don't know, but I will have a look at this too if I fond time.

**Comment by samratch** [comment]
I am also facing the exact same issue, do we have any work around till we get this fixed?

**Comment by samratch** [comment]
however when i install yo i get the .bowerrc error, 'yo doctor' command reproduces it whenever we want.
![yo-error](https://cloud.githubusercontent.com/assets/16188308/19416868/c2a8599c-93b9-11e6-8dd0-78eebd7f0123.jpg)

**Comment by neloaguila** [comment]
So far, I still can't find a way to deal with this, but, I haven't checked if the issue is still there for the current versions of node-webkit, yo and npm (if they've been updated). Basically, I'm waiting for @mschaaf if he has found something.

**Comment by andmart** [comment]
Same problem here.
Node v4.4.3 || 6.10.0
Npm 4.0.2 || 4.4.1
Yo 1.6.0 || 1.8.5
